src/main/java/ttt/Board.java

```diff
@@ -0,0 +1,150 @@
+package ttt;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+public class Board {
+
+    private String[] grid = new String[9];
+    private String status;
+    private String emptyCell;
+    private String winningMark;
+
+    public Board() {
+        this.emptyCell = " ";
+        this.winningMark = null;
+        for (int cell = 0; cell < 9; cell++) {
+            grid[cell] = emptyCell;
+        }
+    }
+
+    public void placeMark(String symbol, int location) {
+        if (!isValidLocation(location)) {
+            this.status = "invalid location";
```

Comment:
I have two suggestions for the 'status' field:
a) get rid of it 😸 
b) use an `enum`.

Getting rid of it will prevent people (you) forgetting to check it after have called `placeMark`. 
All you need to do is change the return type of the method to something other than `void`.

Using an `enum` to indicate the `state` has the advantage of being more restrictive and thus precise.
An `enum` is something between a class and an instance. It can only take the very specific values you create it for. In your case, you could do:
```
public enum State {
  INVALID_LOCATION,
  TAKEN,
  MARK_PLACED
}```
While you are currently expressing the same thing with your Strings, it is possible for your to miss-type it (an extra whitespace somewhere, or maybe a capital letter) and your application will have some error when you run it. With the enum, it has to be exact, or the compiler (!) will complain.